core/new-gui/src/app/workspace/service/execute-workflow/execute-workflow.service.ts

```diff
@@ -88,6 +101,7 @@ export class ExecuteWorkflowService {
         // backend will either respond an execution result or an error will occur
         // handle both cases
         response => {
+
```

Comment:
currently `updateResultMap` is called in result panel component, change it to update the result map here. 
 
Also make `updateResultMap` a private function, and make `getResultMap` returns a `ReadonlyMap` instead of a `Map`. 

In most cases it's good to only expose immutable data structures, and we should try to keep mutations local.